managed/src/main/java/com/yugabyte/yw/cloud/aws/AWSCloudImpl.java

```diff
@@ -82,4 +87,28 @@ private static AWSCredentialsProvider getCredsOrFallbackToDefault(String accessK
       .collect(groupingBy(InstanceTypeOffering::getInstanceType,
         mapping(InstanceTypeOffering::getLocation, toSet())));
   }
+
+  @Override
+  public boolean isValidCreds(Map<String, String> config, String region) {
+    AWSCredentialsProvider credentialsProvider = getCredsOrFallbackToDefault(
+      config.get("AWS_ACCESS_KEY_ID"),
+      config.get("AWS_SECRET_ACCESS_KEY"));
+    try {
+      AmazonEC2 ec2Client = AmazonEC2ClientBuilder
+        .standard()
+        .withCredentials(credentialsProvider)
+        .withRegion(Regions.valueOf(region.toUpperCase()))
+        .build();
+      DryRunResult<DescribeInstancesRequest> dryRunResult = ec2Client
+        .dryRun(new DescribeInstancesRequest());
```

Comment:
Is testing whether the user can describe instances sufficient? Or do we want the pre-flight checks to comprehensively do a dry run of a set of actions such that it covers the list of required IAM permissions needed for a cloud provider?

Also you can just re-use the "getEcC2Client" method by creating local models of provider + region from the request payload without saving these models to the DB and then passing them to "getEcC2Client".

Reply:
Actually both of these were my suggestions in first review :)

Reply:
Basically we need a framework where if you add request Foo - we can easily add a dry run for that in here. 
But I felt this is a good start and more can be added later. 
I was planning to hold the line on getEcC2Client. You should switch to using that.
So either do what daniel suggested 
OR
Another way is to extract getEC2ClientInternal or something to so it becomes usable from getEc2Client and from here.

Reply:
@sb-yb @daniel-yb  Yup  I think more can be added later. For now, I am keeping this as it is(Stick to only DescribeInstancesRequest). 
Not really sure on how to ensure the list of required IAM permissions needed for a cloud provider. Please let me know if its ok else would like to discuss about this. 

Regarding to use getEcC2Client, here we have provider config but I will have to form Region to fit region details in it and then pass in the param - Need to figure out how, will update the PR with this changes shortly. Thanks!

Reply:
Answered above.